reactapp/src/Assignment2/Question16/EmployeeSalary.jsx

```diff
@@ -0,0 +1,67 @@
+//Question16
+// Create a functional component named EmployeeSalary that displays the average salary of a list of employees.
+// Define an array of employee objects, where each object has a name and salary property.
+// Use the useMemo hook to calculate the average salary of employees.
+// Ensure that the useMemo hook has a dependency on the employee data so that it recalculates when the employee data changes.
+// Render the average salary on the screen.
+// Include a button that, when clicked, updates the employee data with new salaries.
+
+
+import React, { useMemo, useState } from 'react'
+
+function EmployeeSalary() {
+  const [name1, setName1] = useState("");
+  const [salary1, setSalary1] = useState("");
+  const [employeeArr, setEmployeeArr] = useState([{ name: "Varun", salary: 25000 },{ name: "Tarun", salary: 40000 }, { name: "Dhoni", salary: 5000 },
+  { name: "Rohit", salary: 6000 },
+  ]);
+
+  const modify = () => {
+    const updatedArr = employeeArr.map((element, index) => {
+      if (element.name === name1) {
+        return { salary: Number(salary1) }; 
+      } else if (element.name !== name1 && index === employeeArr.length - 1) {
+        return { name: name1, salary: Number(salary1) };
+      } else return element; 
+    });
```

Comment:
because you are returning from each condition, you don't need else
example:
if(name === name1) return {salary: salary1}
if(index === lastIndex) return {name, salary}
return element